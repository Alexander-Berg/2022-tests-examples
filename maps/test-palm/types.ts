import {Page} from 'puppeteer';

enum Platform {
    DESKTOP = 'desktop'
};

type BrowserCommands = Omit<
    Page,
    | 'executionContext'
    | 'desiredCapabilities'
    | 'sessionId'
    | 'options'
    | 'isIOS'
    | 'isAndroid'
    | 'isPhone'
    | 'then'
    | 'waitForVisible'
> & {
    waitForVisible: (selector: string, milliseconds?: number, reverse?: boolean) => Promise<null>;
};

type TestCommandName = keyof BrowserCommands;

type BrowserStub<T> = {
    [K in TestCommandName]?:
        | null
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
        | ((...args: BrowserCommands[K] extends (...args: any) => any ? Parameters<BrowserCommands[K]> : never[]) => T);
} & {
    isPhone: boolean;
};

type GetBrowserStub<T, A extends Record<string, unknown> = {}> = (platform: Platform) => BrowserStub<T> & A;

export {Platform, BrowserStub, GetBrowserStub};
