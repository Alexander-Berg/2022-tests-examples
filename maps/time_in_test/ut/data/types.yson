{
    "id"="type-1";
    "key"="bug";
    "version"=13452345;
};
{
    "id"="type-2";
    "key"="task";
    "version"=13452345;
};
{
    "id"="type-3";
    "key"="feature";
    "version"=13452345;
};
