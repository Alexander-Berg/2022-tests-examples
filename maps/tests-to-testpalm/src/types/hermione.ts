/// <reference types='hermione' />

interface ExcludeReason {
    (reason?: string): never;

    in: (match: RegExp | string, reason: string) => never;
}

interface HermioneContext {
    browser: WebdriverIO.Browser;
    skip: ExcludeReason;
    only: ExcludeReason;
}

type HermionePerformCommand = (
    this: WebdriverIO.Browser,
    callback: () => void | Promise<void>,
    description: string | null
) => void | Promise<void>;

interface MockExcludeReason {
    in: (match: RegExp | string, reason: string) => void;
}

interface MockHermioneContext {
    skip: MockExcludeReason;
    only: MockExcludeReason;
    testPalm: {
        setComponent: (component: string) => void;
    };
}

export {HermioneContext, HermionePerformCommand, MockHermioneContext};
