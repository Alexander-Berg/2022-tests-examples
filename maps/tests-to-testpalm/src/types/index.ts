import {TestCaseAttributes, TestCaseProperty, TestCase, TestCaseStep} from '@yandex-int/testpalm-api';
import type {MockHermioneContext} from './hermione';

type Without<T, U> = {[P in Exclude<keyof T, keyof U>]?: never};

type XOR<T1, T2> = T1 | T2 extends object ? (Without<T1, T2> & T2) | (Without<T2, T1> & T1) : T1 | T2;

enum Engine {
    HERMIONE = 'hermione',
    PUPPETEER = 'puppeteer'
}

interface BrowserDescriptionStub {
    [key: string]: unknown;
    retrieveCommands: () => string[];
    getCommandsWithoutDescription: () => Set<string>;
}

interface BrowserAttributesStub {
    [key: string]: unknown;
    retrieveAttributes: () => TestCaseAttributes;
}

interface BrowserEstimateStub {
    [key: string]: unknown;
    retrieveEstimate: () => number;
    getCommandsWithoutEstimate: () => Set<string>;
}

interface BrowserPreconditionsStub {
    [key: string]: unknown;
    retrievePreconditionsData: () => string;
}

type Stub = XOR<XOR<XOR<BrowserDescriptionStub, BrowserAttributesStub>, BrowserEstimateStub>, BrowserPreconditionsStub>;

interface SuiteCallback<T = undefined> {
    (this: T): void | Promise<void>;
}

interface TestCallback<T = undefined> {
    (this: T): void | Promise<void>;
}

interface SuiteFunction {
    (title: string, callback: SuiteCallback): void;
    only?: (title: string, callback: SuiteCallback) => void | Promise<void>;
    skip?: (title: string, callback: SuiteCallback) => void | Promise<void>;
}

interface TestFunction {
    (title: string, callback: TestCallback): void;
    only?: (title: string, callback: TestCallback) => void | Promise<void>;
    skip?: (title: string, callback: TestCallback) => void | Promise<void>;
}

type HookFunction = (callback: TestCallback) => void;

type SetComponentCommand = (component: string) => void;

type TestsGlobal = Omit<NodeJS.Global, 'beforeEach' | 'afterEach' | 'describe' | 'it' | 'test'> & {
    describe?: SuiteFunction;
    it?: TestFunction;
    test?: TestFunction;

    beforeEach?: HookFunction;
    afterEach?: HookFunction;
} & XOR<{page: Stub}, {hermione: MockHermioneContext}>;

interface TestCaseMeta {
    attributes?: TestCaseAttributes;
    properties?: TestCaseProperty[];
    isAutotest?: boolean;
}

interface SkipNextData {
    skipped?: boolean;
    skipIn?: string;
    skipReason?: string;
}

interface ItRegisteredData extends TestCaseMeta {
    name: string;
    itCallback: TestCallback;
    beforeEachCallbacks: TestCallback[];
    afterEachCallbacks: TestCallback[];
}

interface DescribeRegisteredData extends SkipNextData {
    name: string;
    its: ItRegisteredData[];
    component: string;
    beforeEachCallback?: TestCallback;
    afterEachCallback?: TestCallback;
}

type LocalTestCase = Partial<TestCase> & {stepsExpects: TestCaseStep[]};

interface LocalTestCasesParseData {
    localTestCases: LocalTestCase[];
    parsingErrors: Error[];
}

type StubValue<T> = ((...args: unknown[]) => T) | boolean | null;

interface BrowserStubs {
    attributes: Partial<Record<string, StubValue<TestCaseAttributes>>>;
    description: Partial<Record<string, StubValue<string>>>;
    estimate: Partial<Record<string, StubValue<number>>>;
    preconditions?: {
        // Use `{{key}}` in tamplate from `browserStub.preconditions`.
        template: string;
    } & Partial<Record<string, string | StubValue<Record<string, string>>>>;
}

interface SelectorDescription {
    description: string;
    screenshot: string | null;
}

interface SelectorsDescription {
    [key: string]: SelectorDescription;
}

type Platform = 'desktop' | 'mobile';

interface Config {
    testPalm: {
        projectId: string;
        token: string;
    };
    browserId: string;
    filePatterns: string[];
    getComponentByFile?: (filePath: string) => string;
    browserStubs: BrowserStubs;
    minEstimate?: number;
    getSelectorsDescriptions: () => SelectorsDescription;
    getTicket: () => Promise<string | undefined>;
    report: {
        folder: string;
        name: string;
    };
    testcop: {
        project: string;
        tool: string;
        branch: string;
    };
    engine: Engine;
    trackerQueue: string;
}

type Mode = 'validation' | 'dry-run' | 'full' | 'pr';

interface RunParams {
    configPath: string;
    mode: Mode;
    issue?: string;
}

export {
    MockHermioneContext,
    TestCaseMeta,
    LocalTestCase,
    TestsGlobal,
    TestCallback,
    SuiteCallback,
    SkipNextData,
    ItRegisteredData,
    DescribeRegisteredData,
    LocalTestCasesParseData,
    BrowserStubs,
    SelectorsDescription,
    SelectorDescription,
    Config,
    Mode,
    RunParams,
    TestCaseAttributes,
    Platform,
    BrowserDescriptionStub,
    BrowserAttributesStub,
    BrowserEstimateStub,
    BrowserPreconditionsStub,
    XOR,
    Stub,
    Engine,
    SuiteFunction,
    TestFunction,
    SetComponentCommand
};
