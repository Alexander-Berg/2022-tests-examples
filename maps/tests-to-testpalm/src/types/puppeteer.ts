type PuppeteerPerformCommand = (callback: () => void | Promise<void>, description?: string) => Promise<void>;

export {PuppeteerPerformCommand};
