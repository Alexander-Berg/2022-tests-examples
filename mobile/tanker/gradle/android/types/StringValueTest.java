package ru.yandex.tanker.gradle.android.types;

import org.fest.assertions.api.Assertions;
import org.junit.Test;

public class StringValueTest {

    @Test
    public void partialEscapeTest() throws Exception {
        assertEscapesAs("\'", "\\'");
        assertEscapesAs("\\'", "\\'");
        assertEscapesAs("Well it's working!", "Well it\\'s working!");
        assertEscapesAs("Well it\\'s working!", "Well it\\'s working!");
        assertEscapesAs("wow'", "wow\\'");
        assertEscapesAs("wow\\'", "wow\\'");
        assertEscapesAs("'m fine", "\\'m fine");
        assertEscapesAs("\\'m fine", "\\'m fine");

        assertEscapesAs("a>b", "a\\>b");
        assertEscapesAs("a\\>b", "a\\>b");

        assertEscapesAs("a<b", "a\\<b");
        assertEscapesAs("a\\<b", "a\\<b");

        assertEscapesAs("a&b", "a\\&b");
        assertEscapesAs("a\\&b", "a\\&b");

        assertEscapesAs("a\"b", "a\\\"b");
        assertEscapesAs("a\\\"b", "a\\\"b");
    }

    private void assertEscapesAs(String in, String res) {
        Assertions.assertThat(StringValue.partialEscape(in)).isEqualTo(res);
    }
}
