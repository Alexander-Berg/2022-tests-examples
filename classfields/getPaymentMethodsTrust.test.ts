import initPaymentMock, { TrustPaymentMethods } from 'auto-core/server/resources/publicApiBilling/methods/initPayment.mock';

import { TWalletRefillService } from 'auto-core/types/TBilling';

import getPaymentMethodsTrust from './getPaymentMethodsTrust';
import getCustomProduct from './getCustomProduct';

it('для привязки карты вернет пустой массив', () => {
    const response = {
        initPayment: initPaymentMock.withPaymentMethods([
            TrustPaymentMethods.tied_card_1,
            TrustPaymentMethods.new_card,
        ]).value(),
    };
    const result = getPaymentMethodsTrust(response, getCustomProduct([ { name: TWalletRefillService.BIND_CARD, count: 10 } ]));

    expect(result).toEqual([]);
});

it('правильно форматирует данные карт', () => {
    const response = {
        initPayment: initPaymentMock
            .withPaymentMethods([
                TrustPaymentMethods.tied_card_1,
                TrustPaymentMethods.tied_card_2,
            ])
            .value(),
    };
    const result = getPaymentMethodsTrust(response);

    expect(result).toMatchSnapshot();
});

describe('оставит в методах только привязанные карты', () => {
    it('при пополнении кошелька', () => {
        const response = {
            initPayment: initPaymentMock.withPaymentMethods([
                TrustPaymentMethods.tied_card_1,
                TrustPaymentMethods.tied_card_2,
                TrustPaymentMethods.new_card,
            ]).value(),
        };
        const result = getPaymentMethodsTrust(response, getCustomProduct([ { name: TWalletRefillService.WALLET_USER, count: 10 } ]));

        expect(result).toHaveLength(2);
        expect('number' in result[0] && result[0].number).toBe('510000|5755');
        expect('number' in result[1] && result[1].number).toBe('510000|1554');
    });

    it('если сумма на кошельке меньше суммы платежа', () => {
        const response = {
            initPayment: initPaymentMock
                .withPaymentMethods([
                    TrustPaymentMethods.tied_card_1,
                    TrustPaymentMethods.tied_card_2,
                    TrustPaymentMethods.new_card,
                ])
                .withAccountBalance('10000')
                .withCost('10100')
                .value(),
        };
        const result = getPaymentMethodsTrust(response);

        expect(result).toHaveLength(2);
        expect('number' in result[0] && result[0].number).toBe('510000|5755');
        expect('number' in result[1] && result[1].number).toBe('510000|1554');
    });
});

it('добавит кошелек в способы оплаты, если суммы на балансе достаточно', () => {
    const response = {
        initPayment: initPaymentMock
            .withPaymentMethods([
                TrustPaymentMethods.tied_card_1,
                TrustPaymentMethods.tied_card_2,
                TrustPaymentMethods.new_card,
            ])
            .withAccountBalance('20000')
            .withCost('10000')
            .value(),
    };
    const result = getPaymentMethodsTrust(response);

    expect(result[0]).toMatchSnapshot();
});
