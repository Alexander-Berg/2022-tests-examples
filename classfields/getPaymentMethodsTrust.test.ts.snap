// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`добавит кошелек в способы оплаты, если суммы на балансе достаточно 1`] = `
Object {
  "methodId": "wallet",
  "methodType": "Custom",
  "name": "Кошелёк 200 ₽",
}
`;

exports[`правильно форматирует данные карт 1`] = `
Array [
  Object {
    "issuer": undefined,
    "methodId": "card-xfef56f4b79cff2b5bd45d912",
    "methodType": "BankCard",
    "number": "510000|5755",
    "paySystem": "Mastercard",
  },
  Object {
    "issuer": "TinkoffBank",
    "methodId": "card-xf84b20b2355444d84a2eb98d",
    "methodType": "BankCard",
    "number": "510000|1554",
    "paySystem": "Visa",
  },
]
`;
